components/ee/ws-scheduler/pkg/scheduler/state.go

```diff
@@ -233,33 +230,57 @@ func GetRequestedRAMForPod(pod *corev1.Pod) res.Quantity {
 	return *requestedRAM
 }
 
+// GetRequestedEphemeralStorageForPod calculates the amount of ephemeral storage requested by all containers of the given pod
+func GetRequestedEphemeralStorageForPod(pod *corev1.Pod) res.Quantity {
+	requestedEphStorage := res.NewQuantity(0, res.BinarySI)
+	for _, c := range pod.Spec.Containers {
+		requestedEphStorage.Add(*c.Resources.Requests.StorageEphemeral())
```

Comment:
Ah, possible nil deref :point_up_2: 

Really helps to review own code in another interface...

Reply:
I thought the same, but then looked at at StorageEphemeral and it [can't return nil](https://github.com/kubernetes/kubernetes/blob/0923b9abce7382888a077860da28a635975a0f40/pkg/apis/core/resource.go#L57)